#!/usr/bin/env python
PACKAGE = "valkyrie_sandbox"
from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("hipx", double_t, 0, "HIP X", 0, -1.0, 1.0)
gen.add("hipy", double_t, 0, "Hip Y", 0, -1.0, 1.0)
gen.add("hipz", double_t, 0, "RAISE", 0.9, -1.0, 1.0)

gen.add("pelvisRoll", double_t, 0, "Rotate", 0, -1.0, 1.0)
gen.add("pelvisPitch", double_t, 0, "Rotate", 0, -1.0, 1.0)
gen.add("pelvisYaw", double_t, 0, "Rotate", 0, -1.0, 1.0)

gen.add("chestRoll", double_t, 0, "Rotate", 0, -1.0, 1.0)
gen.add("chestPitch", double_t, 0, "Rotate", 0, -1.0, 1.0)
gen.add("chestYaw", double_t, 0, "Rotate", 0, -1.0, 1.0)


gen.add("rightJ1", double_t, 0, "RightArmj0", 0, -2.0, 2.0)
gen.add("rightJ2", double_t, 0, "RightArmj0", 0, -2.0, 2.0)
gen.add("rightJ3", double_t, 0, "RightArmj0", 0, -2.0, 2.0)
gen.add("rightJ4", double_t, 0, "RightArmj0", 0, -2.0, 2.0)
gen.add("rightJ5", double_t, 0, "RightArmj0", 0, -2.0, 2.0)
gen.add("rightJ6", double_t, 0, "RightArmj0", 0, -2.0, 2.0)
gen.add("rightJ7", double_t, 0, "RightArmj0", 0, -2.0, 2.0)

gen.add("leftJ1", double_t, 0, "LeftArmj0", 0, -2.0, 2.0)
gen.add("leftJ2", double_t, 0, "LeftArmj0", 0, -2.0, 2.0)
gen.add("leftJ3", double_t, 0, "LeftArmj0", 0, -2.0, 2.0)
gen.add("leftJ4", double_t, 0, "LeftArmj0", 0, -2.0, 2.0)
gen.add("leftJ5", double_t, 0, "LeftArmj0", 0, -2.0, 2.0)
gen.add("leftJ6", double_t, 0, "LeftArmj0", 0, -2.0, 2.0)
gen.add("leftJ7", double_t, 0, "LeftArmj0", 0, -2.0, 2.0)

exit(gen.generate(PACKAGE, "valkyrie_sandbox", "valkyrie"))
